File: Data Structures & Algorithms/reverse-string/submission-2.java
class Solution {
    public void reverseString(char[] s) {
        int length = s.length;

        for(int i=0;i<length/2;i++){
            char a = s[i];
            char b = s[length- 1 -i];

            if(a != b){
                s[i] = b;
                s[length-1-i] = a;
            }
        }
    }


}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prefix = [1] * len(nums)
        suffix = [1] * len(nums)
        sol = [1] * len(nums)

        pre = 1
        for i in range(len(nums)):
            prefix[i] = pre
            pre = pre * nums[i]

        post = 1
        for i in range(len(nums)-1,-1,-1):
            suffix[i] = post
            post = post * nums[i]
        
        for i in range(len(nums)):
            sol[i] = prefix[i] * suffix[i]
        
        return sol
        

